driver/main.c

```diff
@@ -2958,12 +2968,7 @@ static int set_g_buffer_bytes_dim(const char *val, const struct kernel_param *kp
 	return param_set_ulong(val, kp);
 }
 
-static const struct kernel_param_ops g_buffer_bytes_dim_ops = {
-	.set	= set_g_buffer_bytes_dim,
-	.get	= param_get_ulong,
-};
-
-module_param_cb(g_buffer_bytes_dim, &g_buffer_bytes_dim_ops, &g_buffer_bytes_dim, 0644);
+module_param_call(g_buffer_bytes_dim, set_g_buffer_bytes_dim, param_get_ulong, &g_buffer_bytes_dim, 0644);
```

Comment:
`module_param_cb` is only present since 2.6.36; instead `module_param_call` is always present (even if it is now "obsolete": https://elixir.bootlin.com/linux/v5.18.19/source/include/linux/moduleparam.h#L296), but they both resolve to same code.

`kstrtoul` was instead added in linux 2.6.38.

Reply:
Yeah under the hood [`module_param_call`](https://elixir.bootlin.com/linux/latest/source/include/linux/moduleparam.h#L297) and [`module_param_cb`](https://elixir.bootlin.com/linux/latest/source/include/linux/moduleparam.h#L175) perform the same stuff